src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -120,6 +120,12 @@ public abstract partial class InMemoryDocumentSessionOperations : IDisposable
         protected internal readonly Dictionary<object, DocumentInfo> DocumentsByEntity =
             new Dictionary<object, DocumentInfo>(ObjectReferenceEqualityComparer<object>.Default);
 
+        /// <summary>
+        /// hold the data required to manage Counters tracking for RavenDB's Unit of Work
+        /// </summary>
+        protected internal readonly Dictionary<string, CountersCache> CountersByDocId =
```

Comment:
Use lazy creation for this, so we won't create a dictionary we won't need most times

Reply:
done